pilot/pkg/model/sidecar.go

```diff
@@ -634,6 +636,46 @@ func (ilw *IstioEgressListenerWrapper) selectVirtualServices(virtualServices []c
 	return importedVirtualServices
 }
 
+// filterVirtualServices selects the virtual services by matching given services' host names.
+func (ilw *IstioEgressListenerWrapper) filterVirtualServices(virtualServices []config.Config, servicesByName map[host.Name]*Service) []config.Config {
```

Comment:
@ramaraochavali This is still a separate function from that one.

Reply:
Yeah. I am wondering how to combine this? Because it is dependent on servicesByPort chosen for that particular RDS. Any suggestions?